File: backend-java/src/main/java/com/kwikpic/model/Photographer.java
package com.kwikpic.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Maps to Mongoose Photographer schema in models/Photographer.js.
 * Mongoose schema uses { timestamps: true }, so createdAt/updatedAt are auto-managed.
 */
@Document(collection = "photographers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Photographer {

    @Id
    private String id;

    @Indexed(unique = true)
    private String email;

    /** BCrypt-hashed password. Field name is "passwordHash" in Mongoose schema. */
    private String passwordHash;

    private String name;

    private String businessName;

    private String avatarUrl;

    /** Password reset OTP and its expiration time */
    private String resetOtp;
    private Date resetOtpExpiry;

    /** ObjectId hex-strings referencing Group documents. */
    private List<String> groups = new ArrayList<>();

    @CreatedDate
    private Date createdAt;

    @LastModifiedDate
    private Date updatedAt;
}

File: backend/src/models/Photographer.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const PhotographerSchema = new Schema({
  email: { type: String, required: true, unique: true },
  passwordHash: { type: String, required: true }, // We will map 'password' to this
  name: { type: String, required: true }, // Made required based on your form needs
  
  avatarUrl: { type: String },
  // --- NEW FIELD ---
  businessName: { type: String, required: true }, 
  // ----------------
  
  groups: [{ type: Schema.Types.ObjectId, ref: 'Group' }]
}, { timestamps: true })

module.exports = mongoose.model('Photographer', PhotographerSchema)